python/cudf/cudf/core/column/string.py

```diff
@@ -1641,16 +1607,14 @@ def index(self, sub, start=0, end=None, **kwargs):
         if end is None:
             end = -1
 
-        if sub == "":
-            result_col = column.as_column(
-                0.0, dtype="float", length=len(self._column)
-            ).set_mask(self._column.mask)
+        result_col = cpp_find(self._column, Scalar(sub, "str"), start, end)
+
+        if self._column.has_nulls:
+            result_col = result_col.astype("float64")
```

Comment:
Why are we typecasting to float64 here?

Reply:
So whenever there is a `NaN` value in the output, pandas return floating point numbers:
```python
>>> s = cudf.Series(['1. Ant.  ', '2. Bee!\n', '3. Cat?\t', None])
>>> s.str.find('\n')
0      -1
1       7
2      -1
3    null
dtype: int32
>>> s.to_pandas().str.find('\n')
0   -1.0
1    7.0
2   -1.0
3    NaN
dtype: float64
```
To obtain similar behavior I'm type-casting to float.

Reply:
We should use null instead of `NaN` in this situation.

Reply:
Removed it and returning `null` instead.